src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/actionplan/GoalEntityMapper.kt

```diff
@@ -36,13 +36,18 @@ abstract class GoalEntityMapper {
    * Maps the supplied [CreateGoalDto] into a new un-persisted [GoalEntity].
    * A new reference number is generated and mapped. The JPA managed fields are not mapped.
    * This method is suitable for creating a new [GoalEntity] to be subsequently persisted to the database.
+   *
+   * `archiveReason` and `archiveReasonOther` are not mapped as they have no corresponding fields in the source `CreateGoalDto` instance
+   * because it is not possible to create a new goal that is already in an archived state.
    */
   @BeanMapping(qualifiedByName = ["addNewStepsDuringCreate"])
   @ExcludeJpaManagedFieldsIncludingDisplayNameFields
   @GenerateNewReference
   @Mapping(target = "createdAtPrison", source = "prisonId")
   @Mapping(target = "updatedAtPrison", source = "prisonId")
-  @Mapping(target = "steps", ignore = true)
+  @Mapping(target = "steps", ignore = true) // Steps are mapped in via the AfterMapping method addNewStepsDuringCreate
```

Comment:
Fixed some mapstruct warning whilst I was here about unmapped properties 👍 

(@james-baker-digital-justice , you'll be glad to know, I agree, there are more annotations than what would be lines of code if these were hand-written. I'm not aiming to do anything about this in this PR, but I have created a story for it - RR-917)